wordpress-mobile/WordPress-iOS
# Fix Reader Discover inconsistent state

This PR aims to fix two different issues:

1. Unsaving items from the "Saved" tab
2. Correctly display a post state on "Reader Discover"

### To test

1. Go to "Reader Discover"
2. Save a post
3. Make sure the bookmark icon is now blue
4. Go to "Saved" tab
5. Unsave the post
6. Go back to "Reader Discover"
7. Make sure the icon is not blue
8. Go back to "Saved" and the post shouldn't be there

## Regression Notes
1. Potential unintended areas of impact
Reader tabs

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested all the reader tabs, closing and reopening the app

3. What automated tests I added (or what prevented me from doing so)
None.

PR submission checklist:
